src/csharp/Grpc.Core/Channel.cs

```diff
@@ -127,6 +127,20 @@ public ChannelState State
             }
         }
 
+        // cached handler for watch connectivity state
+        static readonly BatchCompletionDelegate WatchConnectivityStateHandler = (success, ctx, state) =>
```

Comment:
I wanted to be 100% sure we get a hold of BatchCompletionDelegate instance so that no new delegate is created.
If I used static method, I think under some circumstances, the compiler can still decide to not cache a static instance of BatchCompletionDelegate and ` new BatchCompletionDelegate` would be called on every invocation. With my current approach, there's no doubt about what actually happens.